Lib/importlib/_bootstrap_external.py

```diff
@@ -1255,19 +1259,20 @@ def set_data(self, path, data, *, _mode=0o666):
             except FileExistsError:
                 # Probably another Python process already created the dir.
                 continue
-            except OSError as exc:
-                # Could be a permission error, read-only filesystem: just forget
-                # about writing the data.
+            except (OSError, ValueError) as exc:
```

Comment:
This should be made narrower rather than broader (i.e. only ignoring `PermissionError`)